modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/renderer/FragmentRendererControllerImpl.java

```diff
@@ -135,7 +136,12 @@ private String _getFragmentEntryContentExceptionMessage(
 
 		sb.append("<div class=\"alert alert-danger m-2\">");
 
-		String errorMessage = "an-unexpected-error-occurred";
+		ResourceBundle resourceBundle = ResourceBundleUtil.getBundle(
+			"content.Language", getClass());
+
+		String errorMessage = LanguageUtil.get(
+			resourceBundle,
+			"the-remote-web-app-is-no-longer-available-or-was-deleted");
```

Comment:
Hey @mateomustapic is this method responsible only for handling remote apps exceptions? From a quick glance I don't see why the message should be targeted for remote apps if the code itself doesn't seem to know that the error originated from a missing remote app. I'm I missing something?

Reply:
I am not fully familiar with how these errors are configured in backend. 
I found the usage of this error message in this java file, so I assumed it was error validation for remote apps.
Please advise if this is not the right place to modify this message or if it the right place, how should this message look like in this case.

Reply:
One way to test this would be:

1. Add the Bookmarks portlet to a page
2. Undeploy the Bookmarks portlet (one way is go to modules/apps/bookmarks and run `gw clean`)
3. Reload the page and check the if the error message is now "the-remote-web-app-is-no-longer-available-or-was-deleted"

If that happens, then this change wouldn't be correct.

From looking at the code my guess is this message is supposed to be generic, as it originates from exceptions thrown by [fragmentRenderer.render()](https://github.com/liferay-frontend/liferay-portal/blob/94773adcf30f8260f4f80ad4cb1307789a2dfbf9/modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/renderer/FragmentRendererControllerImpl.java#L100). I'd have to debug to see what the actual thrown exception is to see if there's a way to improve the message. For example

```java
try {
    fragmentRenderer.render()
}
catch (Exception exception) {
    if (exception instanceof UndeployedPortletException) {
        // show message "the portlet is no longer available"
    }
    else if (exception instanceof RenderPortletException) {
        // show message "there was a problem rendering the portlet"
    }
    else {
        // show message "unexpected error"
    }
}
```

Those exceptions are just for the purpose of illustration, I'm not sure what are the actual exceptions that could possibly be thrown there.

Reply:
This seems by all means expected behaviour as @brunobasto mentions.

**Test Steps**
- Add RSS Publisher Widget
- Go to a terminal and run `telnet localhost 11311` (assuming you have gogo shell enabled)
- Execute `lb rss` (locate 1170|Active     |   15|Liferay RSS Web)
- Execute `stop 1170` (with the proper bundle id)

### Result in a Content Page
<img width="1161" alt="Screen Shot 2020-09-01 at 16 10 21" src="https://user-images.githubusercontent.com/905006/91861966-ae6a1680-ec6d-11ea-8b55-468f37a08175.png">

### Result in a Widget Page
<img width="888" alt="Screen Shot 2020-09-01 at 16 07 02" src="https://user-images.githubusercontent.com/905006/91861954-ab6f2600-ec6d-11ea-88ef-bedaee11ba50.png">

In both screenshots the 1st message belongs to a removed Remote App and the second one to the stopped RSS Widget.

Reply:
Just to add 1c more, I think it probably won't be straightforward to add a message specific for Remote Apps given that Fragments are a generic concept and it shouldn't depend on the things it renders. Unless its API provides a channel for the things being rendered to pass on error messages, this message needs to be generic for now. Maybe someone from the Fragments team could provide a better input.